android-stellar-sdk/src/main/java/org/stellar/sdk/requests/StreamHandler.java

```diff
@@ -37,52 +29,54 @@ public StreamHandler(TypeToken<T> type) {
     this.type = type;
   }
 
-  public ServerSentEvent handleStream(final URI uri, final EventListener<T> listener) throws IOException {
-    try {
-      Request request = new Request.Builder()
-          .url(uri.toURL())
-          .build();
+  public ServerSentEvent handleStream(final URI uri, final EventListener<T> listener) {
+    Request request = new Request.Builder()
+        .url(uri.toString())
+        .build();
 
-      return okSse.newServerSentEvent(request, new ServerSentEvent.Listener() {
-        @Override
-        public void onOpen(ServerSentEvent sse, okhttp3.Response response) {
-        }
-
-        @Override
-        public void onMessage(ServerSentEvent sse, String id, String event, String data) {
-          if (data.equals("\"hello\"")) {
-            return;
-          }
-          T object = GsonSingleton.getInstance().fromJson(data, type.getType());
-          listener.onEvent(object);
-        }
-
-        @Override
-        public void onComment(ServerSentEvent sse, String comment) {
-        }
+    return okSse.newServerSentEvent(request, new ServerSentEvent.Listener() {
+      @Override
+      public void onOpen(ServerSentEvent sse, okhttp3.Response response) {
+      }
 
-        @Override
-        public boolean onRetryTime(ServerSentEvent sse, long milliseconds) {
-          return false;
+      @Override
+      public void onMessage(ServerSentEvent sse, String id, String event, String data) {
+        if (data.equals("\"hello\"")) {
```

Comment:
why hello? can you document it by using const?

Reply:
it's not documenting, but seems that's the opening message (comes with event = open and retry value), I can even try to move it to the lower level instead at the listener